WalletWasabi.Daemon/Config.cs

```diff
@@ -0,0 +1,195 @@
+
+	private EndPoint GetEffectiveEndPoint(
+		EndPoint valueInConfigFile,
+		[CallerArgumentExpression(nameof(valueInConfigFile))] string key = "") =>
+		GetEffectiveValue(
+			valueInConfigFile,
+			x =>
+			{
+				if (EndPointParser.TryParse(x, 0, out var endpoint))
+				{
+					return endpoint;
+				}
+
+				throw new ArgumentNullException(key, "Not a valid endpoint");
+			},
+			Args,
+			key);
+
+	private bool GetEffectiveBool(
+		bool valueInConfigFile,
+		[CallerArgumentExpression(nameof(valueInConfigFile))] string key = "") =>
+		GetEffectiveValue(
+			valueInConfigFile,
+			x => x switch
+			{
+				_ when x.Equals("true", StringComparison.InvariantCultureIgnoreCase) => true,
+				_ when x.Equals("false", StringComparison.InvariantCultureIgnoreCase) => false,
+				_ => throw new ArgumentNullException(key, "must be 'true' or 'false'")
+			},
+			Args,
+			key);
+
+	private string GetEffectiveString(
+		string valueInConfigFile,
+		[CallerArgumentExpression(nameof(valueInConfigFile))] string key = "") =>
+		GetEffectiveValue(valueInConfigFile, x => x, Args, key);
+
+	private string? GetEffectiveOptionalString(
+		string? valueInConfigFile,
+		[CallerArgumentExpression(nameof(valueInConfigFile))] string key = "") =>
+		GetEffectiveValue(valueInConfigFile, x => x, Args, key);
+
+	private T GetEffectiveValue<T>(
+		T valueInConfigFile,
+		Func<string, T> converter,
+		[CallerArgumentExpression(nameof(valueInConfigFile))] string key = "") =>
+		GetEffectiveValue(valueInConfigFile, converter, Args, key);
+
+	private static string GetString(
+		string valueInConfigFile,
+		string[] args,
+		[CallerArgumentExpression(nameof(valueInConfigFile))] string key = "") =>
+		GetEffectiveValue(valueInConfigFile, x => x, args, key);
+
+	private static T GetEffectiveValue<T>(
+		T valueInConfigFile,
+		Func<string, T> converter,
+		string[] args,
+		[CallerArgumentExpression(nameof(valueInConfigFile))] string key = "")
+	{
+		key = key.StartsWith(nameof(PersistentConfig) + ".")
+			? key.Remove(0, nameof(PersistentConfig).Length + 1)
+			: key;
+
+		var cliArgKey = ("--" + key + "=");
+		var cliArgOrNull = args.FirstOrDefault(a => a.StartsWith(cliArgKey, StringComparison.InvariantCultureIgnoreCase));
+		if (cliArgOrNull is { } cliArg)
+		{
+			return converter(cliArg[cliArgKey.Length..]);
+		}
+
+		var envKey = "WASABI-" + key.ToUpperInvariant();
+		var envVars = Environment.GetEnvironmentVariables();
+		if (envVars.Contains(envKey))
+		{
+			if (envVars[envKey] is string envVar)
+			{
+				return converter(envVar);
+			}
+		}
+
+		return valueInConfigFile;
+	}
+
+	public EndPoint GetBitcoinP2pEndPoint()
+	{
+		if (Network == Network.Main)
+		{
+			return MainNetBitcoinP2pEndPoint;
+		}
+		if (Network == Network.TestNet)
+		{
+			return TestNetBitcoinP2pEndPoint;
+		}
+		if (Network == Network.RegTest)
+		{
+			return RegTestBitcoinP2pEndPoint;
+		}
+		throw new NotSupportedNetworkException(Network);
```

Comment:
Here and below you can use `switch` like in `GetCoordinatorUri`.

Reply:
This code was moved and is out of the scope of this PR.